pyvirtualcam/native_macos/main.mm

```diff
@@ -37,10 +37,10 @@
 static uint32_t cam_fps_den;
 
 
-static void YfromRGB(uint8_t* y, uint8_t r, uint8_t g, uint8_t b) {
+static void YfromRGB(uint8_t* y, double r, double g, double b) {
     *y = (uint8_t)( 0.257 * r + 0.504 * g + 0.098 * b +  16);
 }
-static void UVfromRGB(uint8_t* u, uint8_t* v, uint8_t r, uint8_t g, uint8_t b) {
+static void UVfromRGB(uint8_t* u, uint8_t* v, double r, double g, double b) {
```

Comment:
This used to be a `double` originally, but I was concerned with performance.
I can think of a couple of common platforms where `double` is much slower than `float` (or fixed point).

Reply:
I see your point, but Macs in general are quite powerful so it's probably not an issue, don't you think? I'm happy to revert it if there is no visible difference in quality.

Reply:
After some experiments on Windows with animated gradients I couldn't notice a quality difference, so I reverted this change again.